gridGeneration.py

```diff
@@ -0,0 +1,259 @@
+    road_counts_his = np.zeros(len(in_grids))
+    # load the Geo-json file and ignore other files
+    if os.path.splitext(in_road)[1] == '.geojson':
+        # open geojson files
+        with open(in_road, encoding='utf-8') as new_f:
+            in_data = json.load(new_f)
+        
+        # process all geometries excluding the 1st one
+        for index in range(len(in_data['features'])):
+            # discard a feature without its feature property
+            if len(in_data['features'][index]['properties']) != 0:
+                road_data.append([in_data['features'][index]['geometry']['type'], in_data['features'][index]['geometry']['coordinates']])
+
+        # calculate the number of counts within a grid giving the road data
+        for index2 in range(len(road_data)):
+            if road_data[index2][0] == 'LineString':
+                np_array = np.array(road_data[index2][1])
+                tmp_array = np.zeros(len(in_grids))
+
+                # iterate through all grids
+                for ind in range(len(in_grids)):
+                    for coordinate_ind in range(np_array.shape[0]):
+                        if in_grids[ind][3] == initial_bb[3] and in_grids[ind][2] == initial_bb[2]:
+                            if (np_array[coordinate_ind, :][0] >= in_grids[ind][0] and np_array[coordinate_ind, :][0] <= in_grids[ind][2] and
+                                np_array[coordinate_ind, :][1] >= in_grids[ind][1] and np_array[coordinate_ind, :][1] <= in_grids[ind][3]):
+                                tmp_array[ind] += 1
+                        elif in_grids[ind][3] == initial_bb[3] and in_grids[ind][2] != initial_bb[2]:
+                            if (np_array[coordinate_ind, :][0] >= in_grids[ind][0] and np_array[coordinate_ind, :][0] < in_grids[ind][2] and
+                                np_array[coordinate_ind, :][1] >= in_grids[ind][1] and np_array[coordinate_ind, :][1] <= in_grids[ind][3]):
+                                tmp_array[ind] += 1
+                        elif in_grids[ind][3] != initial_bb[3] and in_grids[ind][2] == initial_bb[2]:
+                            if (np_array[coordinate_ind, :][0] >= in_grids[ind][0] and np_array[coordinate_ind, :][0] <= in_grids[ind][2] and
+                                np_array[coordinate_ind, :][1] >= in_grids[ind][1] and np_array[coordinate_ind, :][1] < in_grids[ind][3]):
+                                tmp_array[ind] += 1
+                        elif in_grids[ind][3] != initial_bb[3] and in_grids[ind][2] != initial_bb[2]:
+                            if (np_array[coordinate_ind, :][0] >= in_grids[ind][0] and np_array[coordinate_ind, :][0] < in_grids[ind][2] and
+                                np_array[coordinate_ind, :][1] >= in_grids[ind][1] and np_array[coordinate_ind, :][1] < in_grids[ind][3]):
+                                tmp_array[ind] += 1
+                
+                #print('temp :', tmp_array)
+                for ind2 in range(len(in_grids)):
```

Comment:
In the last chunk of code, we've had `index`, `index2`, `ind`, and `ind2`, and it's a little hard to keep track of everything. Could we split this up a little bit or rename some of these variables to be a little more descriptive?

Reply:
updated the variables to descriptive names.